File: tools/inject-theme.py
import os
import re

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    if 'js/theme-init.js' in content:
        return

    # Find the relative path from the themes.css link
    match = re.search(r'href="([^"]*)css/themes\.css"', content)
    if not match:
        return

    relative_prefix = match.group(1)
    script_tag = f'<script src="{relative_prefix}js/theme-init.js"></script>'

    if '<head>' in content:
        content = content.replace('<head>', f'<head>\n    {script_tag}', 1)
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Injected into {filepath}")

for root, dirs, files in os.walk('.'):
    if 'node_modules' in root or '.git' in root or 'tools' in root:
        continue
    for file in files:
        if file.endswith('.html'):
            process_file(os.path.join(root, file))

File: tools/inject-theme.js
const fs = require('fs');
const path = require('path');

function processDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            if (!fullPath.includes('node_modules') && !fullPath.includes('.git') && !fullPath.includes('tools')) {
                processDir(fullPath);
            }
        } else if (fullPath.endsWith('.html')) {
            injectScript(fullPath);
        }
    }
}

function injectScript(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Check if already injected
    if (content.includes('js/theme-init.js')) return;
    
    // Find the relative path from the themes.css link
    const cssMatch = content.match(/href="([^"]*)css\/themes\.css"/);
    if (!cssMatch) return;
    
    const relativePrefix = cssMatch[1]; // e.g., "../" or ""
    const scriptTag = `<script src="${relativePrefix}js/theme-init.js"></script>`;
    
    // Inject right before </head> or right after <head>
    if (content.includes('<head>')) {
        content = content.replace('<head>', `<head>\n    ${scriptTag}`);
        fs.writeFileSync(filePath, content, 'utf8');
        console.log(`Injected into ${filePath}`);
    }
}

processDir(path.join(__dirname));
